intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/inspection/ApolloMissingGraphQLDefinitionImportInspection.kt

```diff
@@ -0,0 +1,171 @@
+import com.apollographql.ijplugin.gradle.gradleToolingModelService
+import com.apollographql.ijplugin.project.apolloProjectService
+import com.apollographql.ijplugin.telemetry.TelemetryEvent
+import com.apollographql.ijplugin.telemetry.telemetryService
+import com.apollographql.ijplugin.util.cast
+import com.apollographql.ijplugin.util.findChildrenOfType
+import com.apollographql.ijplugin.util.findPsiFileByUrl
+import com.apollographql.ijplugin.util.quoted
+import com.apollographql.ijplugin.util.unquoted
+import com.intellij.codeInsight.intention.preview.IntentionPreviewInfo
+import com.intellij.codeInsight.intention.preview.IntentionPreviewUtils
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.lang.jsgraphql.ide.config.GraphQLConfigProvider
+import com.intellij.lang.jsgraphql.psi.GraphQLArrayValue
+import com.intellij.lang.jsgraphql.psi.GraphQLDirective
+import com.intellij.lang.jsgraphql.psi.GraphQLElement
+import com.intellij.lang.jsgraphql.psi.GraphQLElementFactory
+import com.intellij.lang.jsgraphql.psi.GraphQLFile
+import com.intellij.lang.jsgraphql.psi.GraphQLNamedElement
+import com.intellij.lang.jsgraphql.psi.GraphQLSchemaDefinition
+import com.intellij.lang.jsgraphql.psi.GraphQLSchemaExtension
+import com.intellij.lang.jsgraphql.psi.GraphQLVisitor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+
+private const val URL_NULLABILITY = "https://specs.apollo.dev/nullability/v0.1"
+private val DIRECTIVES_TO_CHECK = setOf("semanticNonNull", "catch", "ignoreErrors")
+
+class ApolloMissingGraphQLDefinitionImportInspection : LocalInspectionTool() {
+  override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+    return object : GraphQLVisitor() {
+      override fun visitDirective(o: GraphQLDirective) {
+        super.visitDirective(o)
+        if (!o.project.apolloProjectService.apolloVersion.isAtLeastV4) return
+        if (o.name !in DIRECTIVES_TO_CHECK) return
+        if (!o.isImported()) {
+          holder.registerProblem(o, ApolloBundle.message("inspection.missingGraphQLDefinitionImport.reportText", ApolloBundle.message("inspection.missingGraphQLDefinitionImport.reportText.directive")), ImportDefinitionQuickFix(ApolloBundle.message("inspection.missingGraphQLDefinitionImport.reportText.directive")))
+        }
+      }
+    }
+  }
+}
+
+private fun GraphQLElement.schemaFiles(): List<GraphQLFile> {
+  val containingFile = containingFile ?: return emptyList()
+  val projectConfig = GraphQLConfigProvider.getInstance(project).resolveProjectConfig(containingFile) ?: return emptyList()
+  return projectConfig.schema.mapNotNull { schema ->
+    schema.filePath?.let { path -> project.findPsiFileByUrl(schema.dir.url + "/" + path) } as? GraphQLFile
```

Comment:
I'm not 100% clear whether AKGP and the GraphQL plugin agree on their definition of what `schemaFiles` is. 

In AKGP, it can be anything:

```kotlin
apollo {
  service("api") {
    schemaFiles.set(listOf(file("foo.bar")))
  }
}
```

Is that going to be picked up by the GraphQL plugin?

Reply:
Yes that should work, AKIJP talks to AKGP to get the schema files (around [here](https://github.com/apollographql/apollo-kotlin/blob/1a0d9099e669cf7540a6df375b0fa48b1cca8adf/intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/gradle/GradleToolingModelService.kt#L249)), and then passes this info to the GraphQL plugin. 

(Well not sure it'll work for a `.bar` file, it may need a recognized extension to work)